packages/backend/hardware-sec.go

```diff
@@ -0,0 +1,26 @@
+package main
+import (
+    "strings"
+    "os/exec"
+    "context"
+    "google.golang.org/grpc/codes"
+    "google.golang.org/grpc/status"
+    pb "github.com/canonical/desktop-security-center/packages/proto"
+    "google.golang.org/protobuf/types/known/emptypb"
+)
+
+func (s *HardwareServer) IsSecureBootEnabled(ctx context.Context, p *emptypb.Empty) (*pb.IsSecureBootEnabledResponse, error) {
```

Comment:
Is there is no doing without the second argument?

Reply:
When you compile your protos, you should get output a file named `dsc_grpc.pb.go` which will have interface definitions in it. What you are doing in this file is implementing the `HardwareServer` Interface on your `HardwareServer` struct so that it can be used by your gRPC server

Go interface implementation are [implicit](https://go.dev/tour/methods/10), so in Go you just need to match the method signatures defined for the trait in that generated file to implement the interface, which includes the empty in this case.

What you can do is swap the  `p *emptypb.Empty` to `_  *emptypb.Empty` as you wont use it for anything.